### PR TITLE
Supporting  STM32F429i-Discovery!

### PR DESCRIPTION
**Pull request by JaredCJR**
Supporting  STM32F429i-Discovery!

The [README here with "porting guide" and "available command"](https://github.com/JaredCJR/mini-arm-os-1/tree/porting_STM32F429/07-Threads) has full explaination!

### CODE REVIEW

File: 07-Threads/os.c [hunk 1]
```diff
@@ -14,7 +14,8 @@ static void delay(volatile int count)
 
 static void busy_loop(void *str)
 {
-	while (1) {
+	while (1) 
+    {
```
Comment (jserv) [inline 07-Threads/os.c]: Why?

**Comment by jserv** [comment]
@JaredCJR : Is it possible to use upstream [CMSIS](https://github.com/ARM-software/CMSIS) via [git-submodule](https://git-scm.com/docs/git-submodule) instead of copying files into the current repository?

**Comment by jserv** [comment]
We can create a new directory named after "08-CMSIS", which represents for the portable version of "07-Threads" and multi-board support.

**Comment by JaredCJR** [comment]
@jserv 
The [CMSIS](https://github.com/ARM-software/CMSIS) you mentioned does not include the vendor specific CMSIS(for peripheral).
In the [REAME](https://github.com/JaredCJR/mini-arm-os-1/tree/porting_STM32F429/07-Threads):Porting Guide/STEP 2. , that [CMSIS in the mbed](https://github.com/mbedmicro/mbed/tree/master/libraries/mbed/targets/cmsis) is much useful than the previous CMSIS.
However,it contains a lot of driver that we don't need.
We only need ".h" for memory map.
And the CMSIS need to remove some "ifndef" in order to avoid further "include" in peripheral part,which vary with vendor's implementation.

**I think the [git-submodule](https://git-scm.com/docs/git-submodule) idea is not possible in this case.**

The full idea I learnd from the [MRI porting guide](https://github.com/adamgreen/mri/blob/master/notes/mri-porting.creole#device_uarth--device_uartc)

I will create new PR for "08-CMSIS" later!

**Comment by jserv** [comment]
Affirmative! The major design principle of `08-CMSIS` is to ease the porting efforts by means of CMSIS and/or mbed infrastructure. We can introduce some scripts to manipulate the headers and corresponding source files for OS HAL adaptation.

**Pull request closed without merging**